test/smtcallback.js

```diff
@@ -152,9 +162,20 @@ tape('SMTCheckerCallback', function (t) {
       st.comment('Collecting ' + sources[i] + '...');
       var source = fs.readFileSync(sources[i], 'utf8');
 
-      if (source.includes('// SMTEngine')) {
-        st.skip('Test requires specific SMTChecker engine.');
-        continue;
+      var engine;
+      var option = '// SMTEngine: ';
+      if (source.includes(option)) {
+        let idx = source.indexOf(option);
+        if (source.indexOf(option, idx + 1) !== -1) {
+          st.skip('SMTEngine option given multiple times.');
+          st.end();
+          return;
+        }
+        let re = new RegExp(option + '(\\w+)');
+        let m = source.match(re);
+        assert(m !== undefined);
+        assert(m.length >= 2);
```

Comment:
If the early return for multiple SMT engines is not hit, couldn't this assert be stricter i.e., `m.length == 2`?

Reply:
I don't think so, this array actually has a bunch of info about the match itself, where the first element is the full match, second is the word we were looking for, third is index and so on.